protocol/communities_messenger_token_permissions_test.go

```diff
@@ -936,3 +936,261 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestViewChannelPermissions()
 	s.Require().Len(response.Messages(), 1)
 	s.Require().Equal(msg.Text, response.Messages()[0].Text)
 }
+
+func (s *MessengerCommunitiesTokenPermissionsSuite) testReevaluateMemberPrivilegedRoleInOpenCommunity(permissionType protobuf.CommunityTokenPermission_Type) {
+	community, _ := s.createCommunity()
+
+	createTokenPermission := &requests.CreateCommunityTokenPermission{
+		CommunityID: community.ID(),
+		Type:        permissionType,
+		TokenCriteria: []*protobuf.TokenCriteria{
+			&protobuf.TokenCriteria{
+				Type:              protobuf.CommunityTokenType_ERC20,
+				ContractAddresses: map[uint64]string{testChainID1: "0x123"},
+				Symbol:            "TEST",
+				Amount:            "100",
+				Decimals:          uint64(18),
+			},
+		},
+	}
+
+	response, err := s.owner.CreateCommunityTokenPermission(createTokenPermission)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+	s.Require().True(response.Communities()[0].HasTokenPermissions())
+
+	waitOnCommunityPermissionCreated := s.waitOnCommunitiesEvent(s.owner, func(sub *communities.Subscription) bool {
+		return sub.Community.HasTokenPermissions()
+	})
+
+	err = <-waitOnCommunityPermissionCreated
+	s.Require().NoError(err)
+
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(community.HasTokenPermissions())
+
+	s.advertiseCommunityTo(community, s.alice)
+
+	var tokenPermission *protobuf.CommunityTokenPermission
+	for _, tokenPermission = range community.TokenPermissions() {
+		break
+	}
```

Comment:
that's magic, lol :exploding_head: 
```suggestion
	s.Require().Len(community.TokenPermissions(), 1)
	tokenPermission := community.TokenPermissions()[0]
```

Reply:
one problem - this is a `map[string]*protobuf.CommunityTokenPermission` where key is unknown... you solution work with array only 😞

Reply:
Can't we use a key then?

Reply:
The key is unknown. This way I just extract the first element from the map with 1 element

Reply:
Right, now the magic makes sense :magic_wand: